natsstreaming/nats_streaming.go

```diff
@@ -67,6 +84,12 @@ func (p *asyncMessageSink) PublishMessages(ctx context.Context, acks chan<- subs
 
 	conn := p.sc
 
+	go func() {
+		e := <-p.connectionLost
```

Comment:
you can just assign to `rerr` here

Reply:
Also it might be better to do this in a select with `ctx.Done()` to avoid potentially leaking goroutines.

Reply:
select sounds reasonable.